### PR TITLE
remove internal self.fill, and keep padding_mode as string or number

### LINKED ISSUE #376: NaN appear in Padding, when using history
**🐛Bug**

Hi  
a small one, but a funny one, related to how the internal change of the padding_mode (from int to string) is not clear enough ... for history !

**To reproduce**

```python
# Your code here
import torchio as tio
s=tio.datasets.Colin27()
t=tio.CropOrPad(target_shape=[182,218,182] ,padding_mode =4)
st=t(s)
st_again=st.history[0](s)
print('what is the padding default? {} not {}'.format(st.t1.data[0,0,0,0],st_again.t1.data[0,0,0,0]))

```

```python-traceback
# Your errors here
what is the padding default? 4.0 not nan
st.history
Out[6]: [Pad(padding=(1, 0, 1, 0, 1, 0), padding_mode=constant)]

```

**Expected behavior**

**Actual behavior**

**TorchIO version**

### PR DESCRIPTION
**Pull request by romainVala**
Fixes #376.

**Description**

I remove internal fill attribute, to keep padding_mode as given in input (a string, or a number for the constant mode)
parse_padding_mode was then only use to check padding_mode strings, I rename it to 
check_padding_mode

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [x] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by fepegar** [comment]
Awesome. I added a test that fails on the `master` branch. Thanks, @romainVala!

**Comment by romainVala** [comment]
great, first try was good !
happy to participate
